### DISCUSSION TITLE
How to force capacitor update to latest webview availabe?

### DISCUSSION THREAD

**Discussion by arasrezaei** [ionic-team/capacitor#3713] [General]
I wonder if there will be a way for those users to update their WebView to latest.
for example, some android users have chrome 60 which is so outdated, is there a way to force capacitor to update itself to latest WebView (86) without forcing user to update it manually?

**Comment by jcesarmobile** [comment]
No, it's not possible, users have to update the System WebView or Chrome (depending on the Android version) from Google Play

**Reply by arasrezaei** [reply]
Thanks for response, but dont you consider this as a weakness of ionic?
I love web and its capabilities which grow fast, we can not depend on user for our app to work properly, why the team dont take a decision to solve this?
Apps made with electron although have big sizes but offer latest features with them, so why is this approach is bad? Some custom rooms even dont have webview on them.

**Reply by danielehrhardt** [reply]
You could write a Plugin that search for the latest available Chrome WebView on the PlayStore and compare it with the Local Version if it is older you could push a Alert to the UI.

**Comment by aparajita** [comment]
> why the team dont take a decision to solve this?

Because it's google's (and the phone vendors') fault for not caring about creating an ecosystem where Android could easily be updated (until recently). As a result, a significant percentage of Android users are using versions that are [quite old](https://www.androidauthority.com/android-version-distribution-748439/). Ionic could not possibly work on such old versions of WebView.

> Apps made with electron although have big sizes but offer latest features with them, so why is this approach is bad?

Because electron runs on the desktop, so it can afford to embed the entire Chromium web engine in the app. Memory and disk storage are not really an issue on the desktop, whereas on a phone — especially the countless inexpensive Android phones out there — memory and storage are at a premium. Embedding a complete web engine adds 20-40MB to the app binary. In addition, until recently [there weren't any good options](https://stackoverflow.com/questions/16604150/embedded-chromium-or-webkit-in-android-app) for embedding a web engine on Android, so I'm not surprised Ionic hasn't tried.
